src/diagnostic/image.ts

```diff
@@ -13,10 +13,7 @@ export function start(): void {
 function handler(error: ErrorEvent): void {
     let target = error.target as HTMLElement;
     if (target && target.tagName === "IMG") {
-        data = {
-            source: (target as HTMLImageElement).src,
-            target: getId(target)
-        };
-        encode(Event.ImageError);
+        data = { source: (target as HTMLImageElement).src, target };
+        schedule(encode.bind(this, Event.ImageError));
```

Comment:
>schedule [](start = 8, length = 8)

Wouldn't it be clearer to actually track the promises for these by element id and then, when it resolves, invoke them?